### PR TITLE
Add integration tests for phone, plus address some issues caught by them

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_verifier.ts [hunk 1]
```diff
@@ -61,9 +61,9 @@ export class RecaptchaVerifier
     private readonly parameters: Parameters = {
       ...DEFAULT_PARAMS
     },
-    auth?: Auth | null
+    auth?: externs.Auth | null
   ) {
-    this.auth = auth || (initializeAuth() as Auth);
+    this.auth = (auth as Auth) || (initializeAuth() as Auth);
```
Comment (yuchenshi) [inline packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_verifier.ts:66]:
```suggestion
    this.auth = (auth || initializeAuth()) as Auth;
```

File: packages-exp/auth-exp/test/integration/flows/phone.test.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  });
+
+  it('anonymous users can link (and unlink) phone number', async () => {
+    const { user } = await signInAnonymously(auth);
+    const { uid: anonId } = user;
+
+    const cr = await linkWithPhoneNumber(user, PHONE_A.number, verifier);
+    const linkResult = await cr.confirm(PHONE_A.code);
+    expect(linkResult.operationType).to.eq(OperationType.LINK);
+    expect(linkResult.user.uid).to.eq(user.uid);
```
Comment (yuchenshi) [inline packages-exp/auth-exp/test/integration/flows/phone.test.ts:100]: Please add an assertion to verify phoneNumber is correctly set, before unlinking it.

File: packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_verifier.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_verifier.ts]: Done

File: packages-exp/auth-exp/test/integration/flows/phone.test.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/test/integration/flows/phone.test.ts:100]: Done

**Pull request merged**